app/coffeescripts/views/conferences/ConferenceView.coffee

```diff
@@ -126,3 +128,83 @@ define [
         else
           window.open(data[0].url)
       )
+
+    deleteRecording: (e) ->
+      e.preventDefault()
+      if !confirm I18n.t("Are you sure you want to delete this recording?")
+        $(e.currentTarget).parents('.inline-block').find('.al-trigger').focus()
+        return
+      $deleteButton = $(e.currentTarget).parent()
+      @toggleActionButton($deleteButton, {state: "processing", action: "delete"})
+      @toggleRecordingLink($deleteButton, {state: "processing"})
+      $.ajaxJSON($deleteButton.data('url') + "/recording", "DELETE", {
+          recording_id: $deleteButton.data("id"),
+        }, (data) =>
+          if data.deleted == "true"
+            @removeRecordingRow($deleteButton)
+            return
+          @ensureActionPerformed($deleteButton, 1, @removeRecordingRow)
+      )
+
+    ensureActionPerformed: ($button, attempt, callback) =>
+      $.ajaxJSON($button.data('url') + "/recording", "GET", {
```

Comment:
Should you not be able to get this from the ENV? why are you grabbing the url from the button? I assume you are grabbing all the recordings again from the course it would be better to grab it from the ENV rather than using jquery to grab the url from a button.

Reply:
Ok, a little bit of context.

The recordings are actually not stored in anyway in Canvas. Their metadata is retrieved from the BBB server with a getRecordings request per conference in the list. Yes, it may be a killer, but it is the way it has been implemented for ages, and it is the best way as you don't want to get in storing in Canvas and then syncing the content. As the recordings live in BBB, all the URLs point really to external links.

Also, every request for managing the recording (right now deleteRecordings but later on may be publishRecordings, updateRecordings and protectRecordings) has to be done through Canvas wher the endpoint is the conference itself, but there must be a reference to the recordingID (which is internal to BBB). 

So, the way join, close and even delete is performed on the conference URL (/courses/1/conferences/20/close), see:
```<a class="btn btn-small close_conference_link" href="/courses/1/conferences/20/close">End</a>```

would be the same way required for managing recordings

```
<div class="ig-button" data-id="162c30feb94c7dc336bfe075427f92f974fffd13-1516998778542" data-action="delete" data-url="/courses/1/conferences/20">
  <a href="#" class="btn btn-small icon-trash delete_recording_link" title="Delete">Delete</a>
</div>
```
As the request is performed with ajax (explanation in the next comment), both the URL and the ID have to be stored somewhere.

The recordingID corresponds to the recording, but the URL corresponds to the conference. It may be better then to store the URL at the conference level, but it is irrelevant if the conference doesn't have recordings. It can also be stored at the button link level, but then we would be repeating the same data for all the other actions (hypothetically, as the may be implemented in the near future)

Reply:
@jfederico This url though still points to canvas yes? or does ```$button.data('url')``` in this context point to the BBB server that is getting the recordings? In either case the comment here is suggesting that there is a better way to grab this URL rather than using jquery to grab the data url.  If in fact we grab this url client side and inject it into the button then thats fine grabbing this way but I figured there was more configuration with this button url.

Reply:
That is correct. But even in that case as ENV has two arrays with conferences (concluded_conferences and current_conferences) I would need to pull the conference id from the upper level using jquery and then lookup for the URL in the arrays. It will result in a less efficient solution in my opinion.

It is doable, and it is actually done when removing the row from the DOM, but is more direct this way.

Should I proceed?